core/src/HIP/Kokkos_HIP.hpp

```diff
@@ -52,6 +52,9 @@ class HIP {
       Impl::ManageStream manage_stream = Impl::ManageStream::no);
   KOKKOS_DEPRECATED HIP(hipStream_t stream, bool manage_stream);
 
+  HIP(const int hip_device, hipStream_t stream,
+      Impl::ManageStream manage_stream = Impl::ManageStream::no);
```

Comment:
For anything before rocm5.6, we do not have access to a function which gets a device id from a stream (that I am aware of), so we need a constructor to take a device id.

Reply:
What about we only provide multi-GPU support for `ROCm` 5.6 and later then?

Reply:
I would be ok with that.